src/views.rs

```diff
@@ -916,6 +931,93 @@ async fn show_high_scores(
     }
 }
 
+fn switch_mode(mode: Mode, delta: i8) -> Option<Mode> {
+    assert!(delta == -1 || delta == 1);
+    let i = Mode::ALL_MODES.iter().position(|m| *m == mode).unwrap() as i8 + delta;
+    return if i < 0 || i >= (Mode::ALL_MODES.len() as i8) {
+        None
+    } else {
+        Some(Mode::ALL_MODES[i as usize])
+    };
+}
+
+pub async fn show_all_high_scores(client: &mut Client) -> Result<(), io::Error> {
+    let (sender, mut receiver) = watch::channel(HighScoresStatus::Loading);
+    tokio::spawn(game_wrapper::handle_loading_all_high_scores(sender));
+
+    let bottom_text_y = 21;
+    let mut mode = Mode::ALL_MODES[0];
+    let mut loading_task_done = false;
+
+    loop {
+        {
+            let mut render_data = client.render_data.lock().unwrap();
+            render_data.clear(80, 24);
+
+            match &*receiver.borrow() {
+                HighScoresStatus::Loaded(results) => {
+                    render_high_scores_table(
+                        &mut render_data.buffer,
+                        0,
+                        mode,
+                        false,
+                        &results[&mode].single_player_results,
+                        None,
+                    );
+                    render_high_scores_table(
+                        &mut render_data.buffer,
+                        11,
+                        mode,
+                        true,
+                        &results[&mode].multiplayer_results,
+                        None,
+                    );
+
+                    if let Some(prev) = switch_mode(mode, -1) {
+                        render_data.buffer.add_text_with_color(
+                            0,
+                            bottom_text_y,
+                            &format!(" << {} ", prev.name()),
+                            Color::YELLOW_FOREGROUND,
+                        );
+                    }
+                    if let Some(next) = switch_mode(mode, 1) {
+                        render_data.buffer.add_text_with_color(
+                            80 - format!(" {} >> ", next.name()).len(),
+                            bottom_text_y,
+                            &format!(" {} >> ", next.name()),
+                            Color::YELLOW_FOREGROUND,
+                        );
+                    }
+                }
+                status => render_exceptional_high_scores_status(&mut render_data.buffer, status),
+            }
+
+            render_data
+                .buffer
+                .add_centered_text(bottom_text_y, "Press Enter to continue...");
+
+            render_data.changed.notify_one();
+        }
+
+        tokio::select! {
+            result = receiver.changed(), if !loading_task_done => {
```

Comment:
I wonder if this would work
```suggestion
            result = receiver.changed(), if result.is_ok() => {
```

Reply:
computer say no